src/test/java/com/jcabi/github/RtStatusTest.java

```diff
@@ -0,0 +1,93 @@
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+ * STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED
+ * OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.jcabi.github;
+
+import com.google.common.base.Optional;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.Matchers;
+import org.junit.Test;
+
+/**
+ * Test case for {@link RtStatus}.
+ * @author Chris Rebert (github@chrisrebert.com)
+ * @version $Id$
+ */
+public final class RtStatusTest {
+    /**
+     * RtStatus can convert itself to JSON.
+     */
+    @Test
+    public void convertsToJson() {
+        final String url = "http://status.jcabi-github.invalid/42";
+        final String success = "Everything is awesome";
+        final String pending = "Kragle is drying...";
+        final String context = "jcabi/github/test";
+        MatcherAssert.assertThat(
+            new RtStatus(
+                Status.State.Success,
+                Optional.of(url),
+                Optional.of(success),
+                Optional.of(context)
+            ).json().toString(),
+            // @checkstyle LineLength (1 line)
+            Matchers.equalTo("{\"state\":\"success\",\"target_url\":\"http://status.jcabi-github.invalid/42\",\"description\":\"Everything is awesome\",\"context\":\"jcabi/github/test\"}")
+        );
+        MatcherAssert.assertThat(
+            new RtStatus(
+                Status.State.Success,
+                Optional.<String>absent(),
+                Optional.of(success),
+                Optional.of(context)
+            ).json().toString(),
+            // @checkstyle LineLength (1 line)
+            Matchers.equalTo("{\"state\":\"success\",\"description\":\"Everything is awesome\",\"context\":\"jcabi/github/test\"}")
+        );
+        MatcherAssert.assertThat(
+            new RtStatus(
+                Status.State.Failure,
+                Optional.of(url),
+                Optional.<String>absent(),
+                Optional.of(context)
+            ).json().toString(),
+            // @checkstyle LineLength (1 line)
+            Matchers.equalTo("{\"state\":\"failure\",\"target_url\":\"http://status.jcabi-github.invalid/42\",\"context\":\"jcabi/github/test\"}")
+        );
+        MatcherAssert.assertThat(
+            new RtStatus(
+                Status.State.Pending,
+                Optional.of(url),
+                Optional.of(pending),
+                Optional.<String>absent()
+            ).json().toString(),
+            // @checkstyle LineLength (1 line)
+            Matchers.equalTo("{\"state\":\"pending\",\"target_url\":\"http://status.jcabi-github.invalid/42\",\"description\":\"Kragle is drying...\"}")
+        );
+    }
```

Comment:
@cvrebert this test is actually testing four thing. Let's break it up accordingly...